core/src/main/java/org/elasticsearch/ingest/IngestService.java

```diff
@@ -25,17 +25,22 @@
 import java.util.List;
 import java.util.Map;
 
+import org.elasticsearch.common.settings.Setting;
 import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.env.Environment;
 import org.elasticsearch.index.analysis.AnalysisRegistry;
 import org.elasticsearch.plugins.IngestPlugin;
 import org.elasticsearch.script.ScriptService;
 import org.elasticsearch.threadpool.ThreadPool;
 
+import static org.elasticsearch.common.settings.Setting.Property;
+
 /**
  * Holder class for several ingest related services.
  */
 public class IngestService {
+    public static final Setting<Boolean> NEW_DATE_FORMAT =
+        Setting.boolSetting("ingest.new_date_format", false, Property.NodeScope, Property.Dynamic, Property.Deprecated);
```

Comment:
@martijnvg I am a bit new to how settings are managed here. Since we want this setting to be applied to all ingest-nodes, not just one...

is it right to leave it NodeScope+Dynamic and implement the `addSettingsUpdateConsumer` and keep the setting-flag up-to-date? Having this be non-dynamic and set per-node sounds dangerous since one ingest-node may do one thing, while another does something else, etc.